book/src/design/circuit/gadgets/ecc/fixed-base-scalar-mul.md

```diff
@@ -0,0 +1,176 @@
+# Fixed-base scalar multiplication
+
+There are $6$ fixed bases in the Orchard protocol:
+- $\mathcal{K}^{\mathsf{Orchard}}$, used in deriving the nullifier;
+- $\mathcal{G}^{\mathsf{Orchard}}$, used in spend authorization;
+- $\mathcal{R}$ base for $\mathsf{NoteCommit}^{\mathsf{Orchard}}$;
+- $\mathcal{V}$ and $\mathcal{R}$ bases for $\mathsf{ValueCommit}^{\mathsf{Orchard}}$; and
+- $\mathcal{R}$ base for $\mathsf{Commit}^{\mathsf{ivk}}$.
+
+## Witness scalar
+In most cases, we multiply the fixed bases by $255-$bit scalars from $\mathbb{F}_q$. We decompose a full-width scalar $\alpha$ into $85$ $3$-bit windows:
+
+$$\alpha = k_0 + k_1 \cdot (2^3)^1 + \cdots + k_{84} \cdot (2^3)^{84}, k_i \in [0..2^3).$$
+
+At the point of witnessing the scalar, we range-constrain each $3$-bit word of its decomposition.
+$$
+\begin{array}{|c|l|}
+\hline
+\text{Degree} & \text{Constraint} \\\hline
+9 & q_\text{witness-scalar-fixed} \cdot \texttt{range\_check}(k_i, 8) = 0 \\\hline
+\end{array}
+$$
+where $\texttt{range\_check}(k_i, \texttt{range}) = k_i \cdot (1 - k_i) \cdots (\texttt{range} - 1 - k_i).$
```

Comment:
This isn't what's actually implemented. The implementation uses `Iterator::fold(1, ...)` which introduces a (superfluous) constant (which is fixed into the constraint polynomial). However, I just noticed that Rust 1.51 stabilised [`Iterator::reduce`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.reduce) which is the same as `fold` but uses the first element of the iterator as the initial value. We should check through the chip impls to use this where it makes sense, such as here (which would bring the impl back in line with this).

Reply:
Yes we should change the implementation, not the doc.

Reply:
The implementation has been updated to match this doc.